### PR TITLE
Provide the invalid value in the number parsing FormatException message

### PR DESCRIPTION
**Pull request by heathbm**
Implementation of api-approved issue: #65834

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=67415) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=67415) before we can accept your contribution.<br/><br/>:x: heathbm [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=67415)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=67415) it.</sub>

File: src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs [hunk 1]
```diff
@@ -2081,15 +2081,15 @@ internal enum ParsingStatus
...
 
         [DoesNotReturn]
-        internal static void ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type = 0) => throw GetException(status, type);
+        internal static void ThrowOverflowOrFormatException(ParsingStatus status, ReadOnlySpan<char> value, TypeCode type = 0) => throw GetException(status, type, value);
 
         [DoesNotReturn]
         internal static void ThrowOverflowException(TypeCode type) => throw GetException(ParsingStatus.Overflow, type);
 
-        private static Exception GetException(ParsingStatus status, TypeCode type)
+        private static Exception GetException(ParsingStatus status, TypeCode type, ReadOnlySpan<char> value = default)
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs:2089]: Why `value = default`?  That suggests there are some call sites that aren't supplying a value, in which case they're going to end up with a strange `The input string '' was not in a correct format.` message.

Comment (heathbm) [reply src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs]: This has now been addressed in my latest commit

**Review by jeffhandley — APPROVED** [review]
Thanks for this contribution, @heathbm!

**Pull request merged**